dotCMS/src/main/java/com/dotcms/util/content/json/PopulateContentletAsJSONUtil.java

```diff
@@ -0,0 +1,398 @@
+    @WrapInTransaction
+    private void processFile(final File taskDataFile) throws IOException {
+
+        Logger.info(this, "Updating records with missing Contentlet as JSON");
+
+        final MutableInt totalUpdateAffected = new MutableInt(0);
+
+        try (final Stream<String> streamLines = Files.lines(taskDataFile.toPath())) {
+
+            streamLines.forEachOrdered(line -> this.processLine(line, totalUpdateAffected));
+
+            if (!this.paramsUpdate.isEmpty()) {
+                this.doUpdateBatch(totalUpdateAffected);
+            }
+        } finally {
+            Logger.info(this, "-- total updates: " + totalUpdateAffected.intValue());
+        }
+
+        Logger.info(this, "Updated records with missing Contentlet as JSON");
+    }
+
+    /**
+     * Processes the given line preparing the params for the batch update.
+     *
+     * @param line Line with the json representation of the contentlet.
+     * @return
+     * @throws JsonProcessingException
+     */
+    private void processLine(final String line, final MutableInt totalInsertAffected) {
+
+        try {
+            var contentlet = ContentletJsonHelper.INSTANCE.get().immutableFromJson(line);
+
+            final Object contentletAsJSON;
+            if (DbConnectionFactory.isPostgres()) {
+                PGobject jsonObject = new PGobject();
+                jsonObject.setType("json");
+                Try.run(() -> jsonObject.setValue(line)).getOrElseThrow(
+                        () -> new IllegalArgumentException("Invalid JSON"));
+                contentletAsJSON = jsonObject;
+            } else {
+                contentletAsJSON = line;
+            }
+
+            this.paramsUpdate.add(new Params(contentletAsJSON, contentlet.inode()));
+
+            // Execute the batch for the updates if we have reached the max batch size
+            if (this.paramsUpdate.size() >= MAX_UPDATE_BATCH_SIZE) {
+                this.doUpdateBatch(totalInsertAffected);
+            }
+        } catch (JsonProcessingException e) {
+            throw new DotRuntimeException("Error processing line", e);
+        }
+    }
+
+    /**
+     * Converts the contentlet to an immutable contentlet and then builds a json representation of it.
+     *
+     * @param contentlet
+     * @return The Contentlet with the json representation attached to it.
+     */
+    private String toJSON(Contentlet contentlet) {
+
+        try {
+            // Converts the given contentlet to an immutable contentlet and then builds a json representation of it.
+            var contentletAsJSON = ContentletJsonHelper.INSTANCE.get().writeAsString(this.contentletJsonAPI.toImmutable(contentlet));
+
+            // I need to have the JSON in a single line, so I can write it into a file
+            return contentletAsJSON.replaceAll("[\\t\\n\\r]", "");
+        } catch (JsonProcessingException e) {
+            throw new DotRuntimeException("Error creating the JSON representation of the Contentlet", e);
```

Comment:
Let's add the contentlet ID to the error msg.

Reply:
Done